solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -89,14 +93,61 @@ public static Health combine(Collection<Health> states) {
     }
   }
 
-  public ClusterStatus(ZkStateReader zkStateReader, ZkNodeProps props) {
+  public ClusterStatus(ZkStateReader zkStateReader, SolrParams params) {
     this.zkStateReader = zkStateReader;
-    this.message = props;
-    collection = props.getStr(ZkStateReader.COLLECTION_PROP);
+    this.solrParams = params;
+    collection = params.get(ZkStateReader.COLLECTION_PROP);
   }
 
   public void getClusterStatus(NamedList<Object> results)
       throws KeeperException, InterruptedException {
+    List<String> liveNodes = null;
+    NamedList<Object> clusterStatus = new SimpleOrderedMap<>();
+
+    boolean includeAll = this.solrParams.getBool(INCLUDE_ALL, true);
+    boolean withLiveNodes =
+        this.solrParams.getBool(LIVENODES_PROP, includeAll) || (collection != null);
```

Comment:
why compare with collection here?

Reply:
Let me know what you think: When cluster status for a collection is fetched, there is a check at some point to `crossCheckReplicaStateWithLiveNodes()`. `liveNodes` would need to be assigned an empty List in the beginning so `crossCheckReplicaStateWithLiveNodes()` does not throw a `NPE`.  

So if `collection!=null` and `liveNodes=false` what would be returned is the list of replicas, but no indication of whether it is on a live node and therefore usable.

Reply:
Okay I understand that crossCheckReplicaStateWithLiveNodes (used by collection processing) depends on liveNodes.  But the client didn't request liveNodes, therefore we shouldn't return it.  I think "withLiveNodes" boolean should be simple/obvious here, as the others.  Instead down below, you could modify the condition that decides to gather liveNodes by also checking if collection processing is needed.  But still; don't set on the response unless "withLiveNodes".